spec/services/cdl/eligible_item_service_spec.rb

```diff
@@ -0,0 +1,159 @@
+                   call_number: "NA203 .G5 1967",
+                   items: [
+                     {
+                       id: 1_666_782,
+                       on_cdl: "Y",
+                       on_reserve: "N",
+                       copy_number: 2,
+                       item_sequence_number: 1,
+                       temp_location: "null",
+                       perm_location: "uesrf",
+                       circ_group_id: 5,
+                       pickup_location_code: "uescirc",
+                       pickup_location_id: 356,
+                       enum: "null",
+                       chron: "null",
+                       barcode: "32101019151941",
+                       item_type: "NoCirc",
+                       due_date: "null",
+                       status: [
+                         "Not Charged"
+                       ]
+                     }
+                   ]
+                 }
+               ]
+             }.to_json,
+                 headers: { "Content-Type" => "application/json" })
+  end
+
+  describe ".item_ids" do
+    context "on_cdl is null" do
+      before do
+        stub_request(:get, "https://bibdata.princeton.edu/#{bib_id}/items")
+          .to_return(status: 200,
+                     body: { uesrf: [
+                       {
+                         holding_id: 1_581_050,
+                         call_number: "NA203 .G5 1967",
+                         items: [
+                           {
+                             id: 1_666_782,
+                             on_cdl: "null",
+                             on_reserve: "N",
+                             copy_number: 2,
+                             item_sequence_number: 1,
+                             temp_location: "null",
+                             perm_location: "uesrf",
+                             circ_group_id: 5,
+                             pickup_location_code: "uescirc",
+                             pickup_location_id: 356,
+                             enum: "null",
+                             chron: "null",
+                             barcode: "32101019151941",
+                             item_type: "NoCirc",
+                             due_date: "null",
+                             status: [
+                               "Not Charged"
+                             ]
+                           }
+                         ]
+                       }
+                     ] }.to_json, headers: {})
+      end
+      let(:bib_id) { "7214786" }
+
+      it "will not error if the on_cdl is null" do
+        expect(described_class.item_ids(source_metadata_identifier: bib_id)).to eq []
+      end
+    end
+
+    context "querying a suppressed bib" do
+      before do
+        stub_request(:get, "https://bibdata.princeton.edu/#{bib_id}/items")
+          .to_return(status: 404,
+                     body: {}.to_json, headers: {})
+      end
+      let(:bib_id) { "11174664" }
+      it "will not error" do
```

Comment:
I think we should use "will return empty array" instead of saying what it will not do.